### ISSUE TITLE
RCS1231 shouldn't trigger when parameter is `params`

### ISSUE THREAD

**Issue by Sella-GH** [JosefPihrt/Roslynator#1769]
### Description
Per C# compiler message [CS1611](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/params-arrays?f1url=%3FappId%3Droslyn%26k%3Dk(CS1611)#method-declaration-rules) `ref` or `out` is not allowed on `params`. `in` isn't mentioned specifically but the compiler error triggers when declaring it on `params`.

### Example
```cs
public static int Create<T>(params ReadOnlySpan<T> values) // <- this triggers RCS1231
{
    HashCode hash = new();
    foreach (T value in values)
    {
        hash.Add(value);
    }

    return hash.ToHashCode();
}
```
